lib/global-styles.php

```diff
@@ -34,98 +44,299 @@ function gutenberg_get_css_vars( $global_styles_branch, $prefix = '', $is_start
 }
 
 /**
- * Function responsible for enqueuing the style that define the global styles css variables.
+ * Returns an array containing the Global Styles
+ * found in a file, or a void array if none found.
+ *
+ * @param string $global_styles_path Path to file.
+ * @return array Global Styles tree.
  */
-function gutenberg_enqueue_global_styles_assets() {
-	if ( ! locate_template( 'experimental-theme.json' ) ) {
-		return;
-	}
-	$default_global_styles_path = dirname( dirname( __FILE__ ) ) . '/experimental-default-global-styles.json';
-	$default_global_styles      = null;
-
-	if ( file_exists( $default_global_styles_path ) ) {
-		$default_global_styles = json_decode(
-			file_get_contents( $default_global_styles_path ),
+function gutenberg_experimental_global_styles_get_from_file( $global_styles_path ) {
+	$global_styles = array();
+	if ( file_exists( $global_styles_path ) ) {
+		$decoded_file = json_decode(
+			file_get_contents( $global_styles_path ),
 			true
 		);
+		if ( is_array( $decoded_file ) ) {
+			$global_styles = $decoded_file;
+		}
 	}
+	return $global_styles;
+}
+
+/**
+ * Returns an array containing the user's Global Styles
+ * or a void array if none.
+ *
+ * @return array Global Styles tree.
+ */
+function gutenberg_experimental_global_styles_get_user() {
+	$global_styles = array();
+	$user_cpt      = gutenberg_experimental_global_styles_get_user_cpt( array( 'publish' ) );
+	if ( array_key_exists( 'post_content', $user_cpt ) ) {
+		$decoded_data = json_decode( $user_cpt['post_content'], true );
+		if ( is_array( $decoded_data ) ) {
+			$global_styles = $decoded_data;
+		}
+	}
+
+	return $global_styles;
+}
+
+/**
+ * Returns the CPT that contains the user's Global Styles
+ * for the current theme or a void array if none found.
+ * It can also create and return a new draft CPT.
+ *
+ * @param array $post_status_filter Filter CPT by post status.
+ *                                  By default, only fetches published posts.
+ * @param bool  $should_create_draft Whether a new draft should be created
+ *                                   if no CPT was found. False by default.
+ * @return array Custom Post Type for the user's Global Styles.
+ */
+function gutenberg_experimental_global_styles_get_user_cpt( $post_status_filter = array( 'publish' ), $should_create_draft = false ) {
+	$user_cpt         = array();
+	$post_type_filter = 'wp_global_styles';
+	$post_name_filter = 'wp-global-styles-' . strtolower( wp_get_theme()->get( 'TextDomain' ) );
```

Comment:
I think as an id of the theme we should use get_stylesheet() and not the 'TextDomain'. I think the database setting the WordPress core stores in the database to identify the current theme is the stylesheet.

Reply:
The [template parts](https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/template-part/index.php#L22) use the `TextDomain` for this, so I thought we may want to use the same mechanism for API consistency. I'm happy to change it to something else, but I guess it makes sense to update template parts as well?